### PR TITLE
C-4825: Add bulk processing support

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

Screen Shot 2022-01-25 at 3 52 36 AM

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [X] Lint rules pass locally
- [X] The code changed/added as part of this pull request has been covered with tests
- [X] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -377,6 +377,45 @@ async function run() {
...
+  const response = await courier.bulk.createJob({
+    message: {
+      event: "RR4NDQ7NZ24A8TKPWVBEDGE15E9A",
+    },
+  });
+  console.log(response);
+
+  // Example: get a job
+  const response = await courier.bulk.getJob({
+    jobId: "1-61efe386-6ff57552409e311b7a1f371f",
```
Comment (drew-y) [inline README.md:392]: Where did the jobId come from? The response?

Comment (drew-y) [inline README.md:392]:
```suggestion
    jobId: response.jobId,
```

Comment (drew-y) [reply README.md:392]: Though you might wanna rename the other reponse variables 😅

File: src/bulk/types.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+export interface ICourierBulkGetJobUsersResponse {
+  items: ICourierBulkMessageUserResponse[];
+  paging: {
+    cursor?: string;
+    more: boolean;
+  };
+}
+
+export interface ICourierClientBulk {
```
Comment (drew-y) [inline src/bulk/types.ts:88]: Just out of curiosity why do we make an interface for this? Will there be multiple implementations?

**Review by drew-y — APPROVED** [review]
Looks good! Just a few minor comments. Nothing worth blocking over.

File: README.md (hunk 1 above)
Comment (tk26) [reply README.md:392]: yea create job gives jobId back

File: src/bulk/types.ts (hunk 2 above)
Comment (tk26) [reply src/bulk/types.ts:88]: maayybe potentially down the road we could have more generic endpoints for different bulk activities. That wasn't my thinking with this change tho, I kept it similar to other places in the SDK so I don't stand out unnecessarily

File: README.md (hunk 1 above)
Comment (tk26) [reply README.md:392]: I'd keep it hardcoded so the example can work in isolation

**Pull request merged**